Observer/OfflinePaymentObserver.php

```diff
@@ -0,0 +1,118 @@
+use Magento\Framework\Event\Observer;
+
+class OfflinePaymentObserver implements ObserverInterface
+{
+    /**
+    * @var \Magento\Framework\App\Config\ScopeConfigInterface
+    */
+    private $_scopeConfig;    
+    
+    /**
+    * @var  \Magento\Framework\Mail\Template\TransportBuilder
+    */
+    private $_transportBuilder;
+    
+    /**
+    * @var  \Omise\Payment\Helper
+    */
+    private $_helper;
+
+    /**
+     * @var \Omise\Payment\Model\Api\Charge
+     */
+    protected $charge;
+    /**
+     * @var \Magento\Framework\View\Asset\Repository
+     */
+    protected $_assetRepo;
+
+    /**
+     * @param \Omise\Payment\Helper\OmiseHelper $helper
+     * @param \Magento\Framework\Mail\Template\TransportBuilder $transportBuilder
+     * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+     * @param \Magento\Framework\View\Asset\Repository $assetRepo
+     * @param \Omise\Payment\Model\Api\Charge $charge
+     */
+    public function __construct(
+        \Omise\Payment\Helper\OmiseHelper $helper,
+        \Magento\Framework\Mail\Template\TransportBuilder $transportBuilder,
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
+        \Magento\Framework\View\Asset\Repository $assetRepo,
+        \Omise\Payment\Model\Api\Charge  $charge
+    ) {
+        $this->_scopeConfig      = $scopeConfig;
+        $this->_helper           = $helper;
+        $this->_transportBuilder = $transportBuilder;
+        $this->_assetRepo = $assetRepo;
+        $this->charge  = $charge;
+    }
+
+    /**
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return $this
+     */
+    public function execute(Observer $observer)
+    {
+        $order   = $observer->getEvent()->getOrder();
+        $payment = $order->getPayment();
+        $emailData     = new \Magento\Framework\DataObject();
+        $charge_id = $payment->getAdditionalInformation('charge_id');
+        $charge = $this->charge->find($charge_id);
+        switch($payment->getAdditionalInformation('payment_type')) {
+            case 'bill_payment_tesco_lotus':
+                // make sure timezone is Thailand.
+                date_default_timezone_set("Asia/Bangkok");
+                $barcodeHtml   = $this->_helper->convertTescoSVGCodeToHTML($payment->getAdditionalInformation('barcode'));
+                $emailTemplate = 'send_email_tesco_template';
+                $validUntil    = date("d-m-Y H:i:s" , strtotime($charge->expires_at));
+                break;
+            case 'paynow':
+                // make sure timezone is Singapore.
+                date_default_timezone_set("Asia/Singapore");
+                $barcodeHtml   = "<img src= '".$charge->source['scannable_code']['image']['download_uri']."'/>";
+                $emailTemplate = 'send_email_paynow_template';
+                $emailData->setData(['banksUrl' => $this->_assetRepo->getUrl('Omise_Payment::images/supportedbanks.png')]);
```

Comment:
Did we rename this file to make it obvious that this is only for PayNow?

Reply:
Aaah I renamed this image file in previous PR. Need to do this again maybe because renamed the PHP file